tools/n64tool.c

```diff
@@ -384,22 +404,37 @@ int main(int argc, char *argv[])
 		return print_usage(argv[0]);
 	}
 
-	/* Pad the output file to the declared size (not including the IPL3 header) */
-	ssize_t num_zeros = declared_size - total_bytes_written;
-
-	if(output_zeros(write_file, num_zeros))
+	/* If the declared size is too small, error out */
+	if(declared_size && declared_size < total_bytes_written)
 	{
-		fprintf(stderr, "ERROR: Couldn't pad %d bytes to %d bytes. ", total_bytes_written, declared_size);
-		fprintf(stderr, "Increase your size argument to fix this.\n\n");
+		fprintf(stderr, "ERROR: Couldn't fit ROM in %d bytes as requested.\n", declared_size);
 		return print_usage(argv[0]);
 	}
 
+	/* Pad the output file to the declared size (not including the IPL3 header) */
+	if(!declared_size)
+		declared_size = MIN_SIZE;
+	if(declared_size > total_bytes_written)
+	{
+		ssize_t num_zeros = declared_size - total_bytes_written;
+
+		if(output_zeros(write_file, num_zeros))
+		{
+			fprintf(stderr, "ERROR: Couldn't pad %d bytes to %d bytes.\n", total_bytes_written, declared_size);
+			return print_usage(argv[0]);
+		}
+	}
+
 	/* Set title in header */
 	fseek(write_file, TITLE_OFFSET, SEEK_SET);
 	fwrite(title, 1, TITLE_SIZE, write_file);
 
 	/* Sync and close the output file */
 	fclose(write_file);
 
+	/* Rename to the final name */
+	if(rename(tmp_output, output) != 0)
+		perror("cannot rename output file");

```

Comment:
This seems to be a"not-OK" scenario; should this `return STATUS_ERROR;`?

Reply:
This error message makes sense to you, but would leave an end-user clueless. I suggest: `"Couldn't rename temporary output file %s to %s", tmp_output, output`